### PR TITLE
Filters context and functionality added

### PR DESCRIPTION
**Pull request by shobit1337**
- Global State for Filters Context using reducer state.
- Implemented Functionalities.

### CODE REVIEW

File: src/components/Filters/Filters.css [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  color: var(--light-color);
+}
+
+.filter-category-open .filter-category-container {
+  height: auto;
+}
+
+.filters-container {
+  position: sticky;
+  top: 96.5px;
```
Comment (simran61) [inline src/components/Filters/Filters.css:44]: try using relative units

Comment (simran61) [inline src/components/Filters/Filters.css:44]: try using relative units

Comment (simran61) [inline src/components/Filters/Filters.css:44]: try using relative units

**Review by simran61 — APPROVED** [review]
Looks good.
Clean code and folder structure

**Review by Tarandeep-s1ngh — APPROVED** [review]
Suggestions:
1) User might not be able to distinguish between the checkbox and radio button.
2) Loader is visible for the whole time under the products.
3) The pagination shows page 2 by default, even when there's no page 2.
4) Scroll to top is not working properly.

Rest everything looks good!
- dividing the code into multiple components
- good practices followed

Great job!

**Pull request merged**